### ISSUE TITLE
Success Criterion 1.3.6 Identify Purpose (Level AAA)

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#538]
#### [1.3.6 Identify Purpose](https://www.w3.org/TR/WCAG22/#identify-purpose)

(Level AAA)

>In content implemented using markup languages, the purpose of [user interface components](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components), icons, and [regions](https://www.w3.org/TR/WCAG22/#dfn-regions) can be [programmatically determined](https://www.w3.org/TR/WCAG22/#dfn-programmatically-determinable).

##### Applying SC 1.3.6 Identify Purpose to Non-Web Documents and Software

This applies directly as written, and as described in [Intent from Understanding Success Criterion 1.3.6](https://www.w3.org/WAI/WCAG22/Understanding/identify-purpose.html#intent).

***
### Glossary term used only by this SC
><dl>
><dt>region</dt>
><dd>perceivable, programmatically determined section of content</dd>
>
>NOTE
>In HTML, any area designated with a landmark role would be a region.
></dl>

***
**For WCAG2ICT consideration:** 
- Possible notes needed because the concept of "regions" and having those be programmatically determinable may not be available for all non-web software and closed functionality (and maybe non-web document) technologies. Unsure if, across the board, programmatically determinable region attributes are available.
- For the definition of "region", would need to provide non-HTML examples of how one might designate a region, if it's possible.
***

**Comment by maryjom** [comment]
On the surface (my previous review of this SC) it seemed this SC would be easy to apply to non-web technologies - at least with the language used in the SC itself. However, after reviewing the understanding document, I am concerned about this SC's applicability and ability to support it outside of the web. I don't know what a document author or a non-web software author has available to them to even implement this SC.  There are no landmark regions, nor implementations of microdata to markup user interface components or programmatically indicate the purpose of icons and other UI components. Even if these exist, does software have the ability for the users to modify icons to other symbols they know? It's easier on the Web because you can replace things in the DOM. Not so easy in non-web documents and software. 

IMO, this shouldn't be applied so perhaps that is what should be said. That it is problematic to apply because existing non-web technologies are likely to not have the infrastructure to support this criterion.

**Comment by maryjom** [comment]
@GreggVan I tend to agree that the definition seems incomplete. In HTML, the elements (<header>, <main>, <nav>, <footer>, aside, <section>, form, and search) map to similar WAI-ARIA landmark regions (banner, main, navigation, contentinfo, complementary, region, form, and search). These region types don't completely identify the purpose which is done more precisely by providing a good label, though the labeling is not part of SC 1.3.6's requirement.

The intent for these elements is to define an overall structure for the content on a web page, and enable a way to navigate to those sections. This SC wants those used so that the UI can be adjusted to help the user focus on certain regions (through highlighting or hiding content), and to allow the user to change the appearance of certain elements (e.g. icons, links, etc.) with symbols that are more familiar to the user. I am not sure that software currently has the capability to allow users to modify the UI in those ways or that platforms have the capability to support such UI modifications in software designed to run on that platform.

So while we can interpret this SC's direct language into requiring programmatic info for the purpose of sections/regions, and providing a programmatic purpose, the intent of the user being able to adjust/customize the UI of non-web software and documents wouldn't be realized until there is document user agent and platform support for such modifications.

**Comment by GreggVan** [comment]
√

but then we still need a definition.     What is the definition of "section"  and can we just put that into the defintion of 'region" in place of the word section.

**Comment by maryjom** [comment]
The TF discussed the results of the [first Level AAA survey](https://www.w3.org/wbs/55145/LevelAAA-group1/results) (Questions 5 and 6) a few of times and further refined the language. The new proposal is in the [second Level AAA survey](https://www.w3.org/wbs/55145/AAA-take-two/), due 12 February. See questions 5 and 6, planned to be discussed on 12 Feb.
